File: Next_Layer_Boost_App-main/app/src/main/java/com/sis/clighteningboost/Api/ApiInterfaceForNodes.java
package com.sis.clighteningboost.Api;

import com.sis.clighteningboost.Models.DecodeBolt112WithExecuteResponse;
import com.sis.clighteningboost.Models.NodesDataWithExecuteResponse;
import com.sis.clighteningboost.Models.REST.RegistrationClientResp;

import java.util.Map;

import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.Response;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Path;

public interface ApiInterfaceForNodes {
    @FormUrlEncoded
    @POST("connect")
        //ok
    Call<ResponseBody> getNodesData(
            @Field("merchantId") String merchantId,
            @Field("merchantBackendPassword") String merchantBackendPassword,
            @Field("boost2faPassword") String boost2faPassword,
            @Field("command") String command);

    @POST("connect")
    Call<Object> getNodes(@Body Map<String, String> body);


    @POST("execute")
    Call<NodesDataWithExecuteResponse> getNodesDataWithExecute(
            @Header("Authorization")String token,
            @Body Map<String, String> body);

    @POST("execute")
    Call<DecodeBolt112WithExecuteResponse> getDecodeBolt112WithExecute(
            @Header("Authorization")String token,
            @Body Map<String, String> body);

    @POST("execute")
    Call<DecodeBolt112WithExecuteResponse> callRefundApiRequestWithExecute(
            @Header("Authorization")String token,
            @Body Map<String, String> body);

    @POST("/pay")
    Call<DecodeBolt112WithExecuteResponse> callPayApiRequestWithExecute(
            @Header("Authorization")String token,
            @Body Map<String, String> body);
}

File: Next_Layer_Boost_App-main/app/src/main/java/com/sis/clighteningboost/Api/ApiInterfaceForNodes.kt
package com.sis.clighteningboost.Api

import okhttp3.ResponseBody
import com.sis.clighteningboost.Models.NodesDataWithExecuteResponse
import com.sis.clighteningboost.Models.DecodeBolt112WithExecuteResponse
import retrofit2.Call
import retrofit2.http.*

interface ApiInterfaceForNodes {
    @FormUrlEncoded
    @POST("connect")
    fun  //ok
            getNodesData(
        @Field("merchantId") merchantId: String?,
        @Field("merchantBackendPassword") merchantBackendPassword: String?,
        @Field("boost2faPassword") boost2faPassword: String?,
        @Field("command") command: String?
    ): Call<ResponseBody?>?

    @POST("connect")
    fun getNodes(@Body body: Map<String?, String?>?): Call<Any?>?

    @POST("execute")
    fun getNodesDataWithExecute(
        @Header("Authorization") token: String?,
        @Body body: Map<String?, String?>?
    ): Call<NodesDataWithExecuteResponse?>?

    @POST("execute")
    fun getDecodeBolt112WithExecute(
        @Header("Authorization") token: String?,
        @Body body: Map<String?, String?>?
    ): Call<DecodeBolt112WithExecuteResponse?>?

    @POST("execute")
    fun callRefundApiRequestWithExecute(
        @Header("Authorization") token: String?,
        @Body body: Map<String?, String?>?
    ): Call<DecodeBolt112WithExecuteResponse?>?

    @POST("/pay")
    fun callPayApiRequestWithExecute(
        @Header("Authorization") token: String?,
        @Body body: Map<String?, String?>?
    ): Call<DecodeBolt112WithExecuteResponse?>?
}